Backend/NoCode.FlowerShop.Application/Flowers/Create/CreateFlowerCommandHandler.cs

```diff
@@ -0,0 +1,33 @@
+using ErrorOr;
+using MediatR;
+using NoCode.FlowerShop.Application.Common.Interfaces.Persistence;
+using NoCode.FlowerShop.Domain;
+
+namespace NoCode.FlowerShop.Application.Flowers.Create;
+
+public sealed class CreateFlowerCommandHandler 
+    : IRequestHandler<CreateFlowerCommand, ErrorOr<CreateFlowerResult>>
+{
+    private readonly IFlowerRepository _flowerRepository;
+
+    public CreateFlowerCommandHandler(IFlowerRepository flowerRepository)
+    {
+        _flowerRepository = flowerRepository;
+    }
+
+    public async Task<ErrorOr<CreateFlowerResult>> Handle(CreateFlowerCommand request, CancellationToken cancellationToken)
+    {
+        List<Error> errors = new();
+
+        if (errors.Count > 0) return errors;
+
+        Flower flowerToCreate = new(
+            name: request.Name, 
+            imageUrl: request.ImageUrl);
+
+        var createdFlower = await _flowerRepository
+            .AddAsync(flowerToCreate, CancellationToken.None);
```

Comment:
```
var createdFlower = await _flowerRepository.AddAsync(flowerToCreate);
```

Reply:
In theory, this call can also fail and throw an exception, but I think we will deal with these issues later in a more systematic manner.